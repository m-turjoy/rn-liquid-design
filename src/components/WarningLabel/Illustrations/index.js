import CompressedGas from './compressedGas'
import Corrosive from './corrosive'
import EnvHazard from './envHazard'
import Explosive from './explosive'
import Flammable from './flammable'
import Harmful from './harmful'
import HealthHazard from './healthHazard'
import Oxidizing from './oxidizing'
import Toxic from './toxic'

export {
  CompressedGas,
  Corrosive,
  EnvHazard,
  Explosive,
  Flammable,
  Harmful,
  HealthHazard,
  Oxidizing,
  Toxic
}
