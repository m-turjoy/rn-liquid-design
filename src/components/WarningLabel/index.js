import React, { Component } from "react";
import { number, string } from "prop-types";
import { View } from "react-native";
import {
	CompressedGas,
	Corrosive,
	EnvHazard,
	Explosive,
	Flammable,
	Harmful,
	HealthHazard,
	Oxidizing,
	Toxic,
} from "./Illustrations";

class WarningLabel extends Component {
	render() {
		const { name, size } = this.props;

		const warningLabel = () => {
			switch (name) {
				case "compressedGas":
					return <CompressedGas size={size} />;
				case "corrosive":
					return <Corrosive size={size} />;
				case "environmentalHazard":
					return <EnvHazard size={size} />;
				case "explosive":
					return <Explosive size={size} />;
				case "flammable":
					return <Flammable size={size} />;
				case "harmful":
					return <Harmful size={size} />;
				case "healthHazard":
					return <HealthHazard size={size} />;
				case "oxidizing":
					return <Oxidizing size={size} />;
				case "toxic":
					return <Toxic size={size} />;
				default:
					return null;
			}
		};

		return <View>{warningLabel()}</View>;
	}
}

WarningLabel.propTypes = {
	name: string.isRequired,
	size: number,
};

WarningLabel.defaultProps = {
	size: 90,
};

export default WarningLabel;
