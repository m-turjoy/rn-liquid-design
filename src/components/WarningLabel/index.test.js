import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import WarningLabel from "./index";

configure({ adapter: new Adapter() });
describe("WarningLabel component", () => {
	describe("Renders correctly", () => {
		test("it renders default WarningLabel", () => {
			const tree = shallow(<WarningLabel name="explosive" />);
			expect(tree).toMatchSnapshot();
		});
	});
});
