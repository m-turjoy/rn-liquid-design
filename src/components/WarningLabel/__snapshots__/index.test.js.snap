// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningLabel component Renders correctly it renders default WarningLabel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WarningLabel
    name="explosive"
    size={90}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Explosive
        size={90}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "size": 90,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Explosive
          size={90}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": 90,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
